iroh-relay/src/client/tls.rs

```diff
@@ -240,22 +245,26 @@ impl MaybeTlsStreamBuilder {
 
         let (mut sender, conn) = hyper::client::conn::http1::handshake(io)
             .await
-            .context(ProxyConnectSnafu)?;
+            .map_err(|err| e!(DialError::ProxyConnect, err))?;
         task::spawn(async move {
             if let Err(err) = conn.with_upgrades().await {
                 error!("Proxy connection failed: {:?}", err);
             }
         });
 
-        let res = sender.send_request(req).await.context(ProxyConnectSnafu)?;
+        let res = sender
+            .send_request(req)
+            .await
+            .map_err(|err| e!(DialError::ProxyConnect, err))?;
         if !res.status().is_success() {
-            return Err(ProxyConnectInvalidStatusSnafu {
-                status: res.status(),
-            }
-            .build());
+            return Err!(DialError::ProxyConnectInvalidStatus {
+                status: res.status()
+            });
         }
 
-        let upgraded = hyper::upgrade::on(res).await.context(ProxyConnectSnafu)?;
+        let upgraded = hyper::upgrade::on(res)
+            .await
+            .map_err(|err| e!(DialError::ProxyConnect, err))?;
```

Comment:
can't we do this with one of the `context` variants?

Reply:
`context` takes a string (or, impl Display) and adds this as context. We can't do without the `e!` macro if we want jump-to-usage on `ProxyConnect` to work and not type `meta: Default::default()` manually. So not sure what `context` should do here?